eipw-lint/tests/lint_markdown_link_eip.rs

```diff
@@ -0,0 +1,190 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use eipw_lint::lints::markdown::LinkEip;
+use eipw_lint::reporters::Text;
+use eipw_lint::Linter;
+
+#[tokio::test]
+async fn eip_number_mismatch() {
+    let src = r#"---
+header: value1
+---
+[EIP-1](./eip-2.md)
+"#;
+
+    let reports = Linter::<Text<String>>::default()
+       .clear_lints()
+       .deny("markdown-link-eip", LinkEip(r"eip-([^.]*)\.md(#.+)?$".to_string()))
+       .check_slice(None, src)
+       .run()
+       .await
+       .unwrap()
+       .into_inner();
+    assert_eq!(
+        reports,
+        r#"error[markdown-link-eip]: link text does not match link destination
+  |
+4 | [EIP-1](./eip-2.md)
+  |
+  = info: link text should match `[EIP|ERC-2]`
```

Comment:
If you can, I'd rather the specific expected text in the message. Maybe something like:

```
error[markdown-link-eip]: link text does not match link destination
  |
4 | [EIP-1](./eip-2.md)
  |
  = help: `use [EIP-2](./eip-2.md)` instead
```

Reply:
I am working on the help hint. I check the link and build the correct version of the text based on the link. Is it correct? So, my hints will look like this:

`[EIP-1](./eip-2.md)  ->  [EIP-2](./eip-2.md)`
`[EIP-1: Foo](./eip-1.md)  ->  [EIP-1](./eip-1.md)`
`[Another Proposal](./eip-1.md)  ->  [EIP-1](./eip-1.md)`
`[EIP-1](./eip-1.md#motivation)  ->  [EIP-1: Motivation](./eip-1.md#motivation)`
`[EIP-2: Hello](./eip-1.md#abstract)  ->  [EIP-1: Abstract](./eip-1.md#abstract)`
`[Another Proposal](./eip-1.md#rationale)  ->  [EIP-1: Rationale](./eip-1.md#rationale)`

Example:
```
error[markdown-link-eip]: link text does not match link destination
  |
4 | [Another Proposal](./eip-1.md#rationale) 
  |
  = help: ` [EIP-1: Rationale](./eip-1.md#rationale)` instead
```

Reply:
That looks great!